backend/tests/unit/pidstore/test_providers/test_texkey_provider.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import mock
+import pytest
+
+from inspirehep.pidstore.errors import CannotGenerateUniqueTexKey
+from inspirehep.pidstore.providers.texkey import InspireTexKeyProvider
+
+
+def test_sanitize():
+    name = "ZażółcićGęśląJaźń_12!@#$ baba-jaga"
+    expected_sanitazed_name = "Zazolcicgeslajazn12baba-jaga"
+    sanitazed_name = InspireTexKeyProvider.sanitize(name)
+    assert sanitazed_name == expected_sanitazed_name
+
+
+def test_get_texkey_first_part_when_authors_les_than_10():
+    data = {"authors": [{"full_name": "Kirk , James T."}, {"Janeway, Kathryn"}]}
+
+    expected_texkey_part = "Kirk"
+    texkey_part = InspireTexKeyProvider.build_texkey_first_part(data)
+    assert texkey_part == expected_texkey_part
+
+
+def test_get_texkey_first_part_when_authors_more_than_10():
+    data = {
+        "authors": [
+            {"full_name": "Kirk , James T."},
+            {"full_name": "Janeway, Kathryn"},
+            {"full_name": "Picard, Jean-Luc"},
+            {"full_name": "Sisko, Benjamin"},
+            {"full_name": "Archer, Jonathan"},
+            {"full_name": "Georgiou, Philippa"},
+            {"full_name": "Lorca, Gabriel"},
+            {"full_name": "Pike, Christopher"},
+            {"full_name": "Sulu, Hikaru"},
+            {"full_name": "La Forge, Geordi"},
+            {"full_name": "Paris, Owen"},
+        ],
+        "collaborations": [{"value": "CMS"}],
+    }
+
+    expected_texkey_part = "Cms"
```

Comment:
preserving capitalization would also be better here

Reply:
:+1: